plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/ext/DockerTerminalModule.java

```diff
@@ -40,10 +45,29 @@ protected void configure() {
 
         Multibinder<String> volumesMultibinder =
                 Multibinder.newSetBinder(binder(), String.class, Names.named("machine.docker.machine_volumes"));
-        if (SystemInfo.isWindows()) {
-            volumesMultibinder.addBinding().toProvider(TerminalServerBindingProviderWinOS.class);
-        } else {
-            volumesMultibinder.addBinding().toProvider(TerminalServerBindingProviderUnix.class);
+        volumesMultibinder.addBinding().toProvider(TerminalVolumeProvider.class).in(Singleton.class);
+    }
+
+    /**
+     * Provides volumes configuration of machine for terminal
+     *
+     * On Windows MUST be locate in "user.home" directory in case limitation windows+docker.
```

Comment:
In this case `On Windows MUST...` will be in the same line as `for terminal` is, what is worth is that it will be the only sentence, i see that you moved this code from different place, but anyway i think you should fix it, as it is broken